src/bms.rs

```diff
@@ -103,7 +109,11 @@ pub fn parse_bms(source: &str) -> BmsOutput {
     let mut warnings: Vec<BmsWarning> = lex_warnings.into_iter().map(BmsWarning::Lex).collect();
 
     // Parse BMS using default RNG and prompt handler
+    #[cfg(feature = "rand")]
     let rng = RandRng(StdRng::from_os_rng());
+    #[cfg(not(feature = "rand"))]
+    let rng = crate::ast::rng::RngMock([BigUint::one()]);
+
```

Comment:
It's not recommended to change behavior of the function by feature flags. If you want to add feature to change randomness, you can add an optional parameter like:
```rs
pub fn parse_bms(source: &str, rng: Option<impl Rng>) -> BmsOutput {
```